GitHub_Learning/Generic/AlertProvider/Application/AlertProviderController.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  AlertProvider.swift
+//  GitHub_Learning
+//
+//  Created by Amin Ashhab on 19/11/2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import Foundation
+import CleanCore
+
+enum ActionStyle {
+    case _default
+    case cancel
+    case destructive
+}
+
+struct AlertAction {
+    let title: String
+    let style: ActionStyle
+    var completion: ((Any) -> (Void))?
```

Comment:
Void (obecne zadny navratovy typ, pokud to neni tuple) nemusis obalovat v zavorkach.
```suggestion
    var completion: ((Any) -> Void)?
```